File: python/4/37.py
import matplotlib.pyplot as plt
from functions4 import *

freqs = word_freq(morphological())
sorted_freqs = sorted(freqs.items(), key=lambda x:(-x[1], x[0]))

x = range(10)
y = [word[1] for word in sorted_freqs[:10]]
labels = [word[0] for word in sorted_freqs[:10]]
plt.bar(x, y, tick_label=labels)
plt.show()

File: ruby/4/37.rb
require './util'
require 'pycall/import'
include PyCall::Import

# MacOSX backend is not usable through pycall...
pyimport 'matplotlib', as: :mp
mp.rcParams[:backend] = 'TkAgg' if mp.rcParams[:backend] == 'MacOSX'

pyimport 'matplotlib.mlab', as: 'mlab'
pyimport 'matplotlib.pyplot', as: 'plt'


sorted_freqs = morphologies.word_freq.sort_by{ |word, count| [-count, word] }

x = (0...10).to_a
y = sorted_freqs[0...10].map{ |_, count| count }
labels = sorted_freqs[0...10].map{ |word, _| word }

plt.bar(x, y, tick_label: labels)
plt.show()
